packages/react/src/components/InlineLoading/InlineLoading.js

```diff
@@ -24,39 +27,48 @@ export default function InlineLoading({
 }) {
   const loadingClasses = classNames(`${prefix}--inline-loading`, className);
   const getLoading = () => {
-    if (success) {
+    if (status === 'error') {
+      return <Error20 className={`${prefix}--inline-loading--error`} />;
+    }
+    if (status === 'finished') {
       setTimeout(() => {
         if (onSuccess) {
           onSuccess();
         }
       }, successDelay);
       return (
-        <svg
+        <CheckmarkFilled16
           className={`${prefix}--inline-loading__checkmark-container ${prefix}--inline-loading__svg`}
-          xmlns="http://www.w3.org/2000/svg"
-          viewBox="0 0 10 10">
-          <polyline
-            className={`${prefix}--inline-loading__checkmark`}
-            points="0.74 3.4 3.67 6.34 9.24 0.74"
-          />
-        </svg>
+        />
+      );
+    }
+    if (status === 'inactive' || status === 'active') {
+      return (
+        <Loading
+          small
+          description={iconDescription}
+          withOverlay={false}
+          active={status === 'active'}
+        />
       );
     }
-    return (
-      <Loading small description={iconDescription} withOverlay={false} active />
-    );
+    return undefined;
   };
   const loadingText = (
     <p className={`${prefix}--inline-loading__text`}>{description}</p>
   );
+  const loading = getLoading();
+  const loadingAnimation = !loading ? (
```

Comment:
would this be identical?

```suggestion
  const loadingAnimation = loading && <div className={`${prefix}--inline-loading__animation`}>{loading}</div>
```